src/WebJobs.Extensions.EasyTables/Extensions/MobileServiceClientExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.WindowsAzure.MobileServices
+{
+    internal static class MobileServiceClientExtensions
+    {
+        // Specifying TableName on the attribute overrides any table name inferred by the object's type. So if a 
+        // TableName has been set, we have to do some private reflection to update the internal tableNameCache.
+        // We will address this in the Mobile Services code to make this scenario supported and then remove this
+        // private reflection.
+        public static void AddToTableNameCache(this IMobileServiceClient client, Type type, string tableName)
+        {
+            // Get Serializer
+            PropertyInfo serializerProperty = client.GetType().GetProperty("Serializer", BindingFlags.NonPublic | BindingFlags.Instance);
+            object serializer = serializerProperty.GetValue(client);
+
+            // Get SerializerSettings
+            PropertyInfo settingsProperty = serializer.GetType().GetProperty("SerializerSettings", BindingFlags.Public | BindingFlags.Instance);
+            MobileServiceJsonSerializerSettings settings = settingsProperty.GetValue(serializer) as MobileServiceJsonSerializerSettings;
+
+            // Get cache
+            FieldInfo tableNameCacheField = settings.ContractResolver.GetType().GetField("tableNameCache", BindingFlags.NonPublic | BindingFlags.Instance);
+            Dictionary<Type, string> tableNameCache = tableNameCacheField.GetValue(settings.ContractResolver) as Dictionary<Type, string>;
+
+            // Update cache
+            tableNameCache[type] = tableName;
+        }
```

Comment:
This is just awful, who's idea was it to do private reflection?? :trollface: 

Seriously though, lets be a bit more defensive here. If any of this reflection fails, it will do so with a hugely nasty error (an impossible to understand null reference or some such). Can you add some basic null checks and just in general try to make sure the method throws a useful exception if something goes wrong?